### ISSUE TITLE
Enable istioctl integration tests in external istiod topology

### ISSUE THREAD

**Issue by frankbu** [istio/istio#35117]
(This is used to request new product features, please visit <https://discuss.istio.io> for questions on using Istio)

**Describe the feature request**

The `istioctl` integration tests have been disabled in the `integ-pilot-istiodremote-tests` job because they are currently failing:

```
--- FAIL: TestWait (3.56s)
--- FAIL: TestVersion (3.39s)
--- FAIL: TestXdsVersion (3.25s)
--- FAIL: TestProxyStatus (3.08s)
--- FAIL: TestXdsProxyStatus (3.36s)
--- FAIL: TestKubeInject (3.24s)
--- FAIL: TestLocality (0.01s)
--- FAIL: TestAnalysisWritesStatus (301.02s)
```

Some (most?) of these tests should be fixed to work with an external control plane and then re-enabled.

**Describe alternatives you've considered**

**Affected product area (please put an X in all that apply)**

[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[x] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

**Comment by frankbu** [comment]
`Istioctl` can be configured to work with an external control plane, after which many of these tests should not fail.
https://preliminary.istio.io/latest/docs/reference/commands/istioctl/#envvars

For example, to manually configure `istioctl` to work with [the documented external control plane setup](https://preliminary.istio.io/latest/docs/setup/install/external-controlplane), I set the following env variables:
```
export ISTIOCTL_PREFER_EXPERIMENTAL=true
export ISTIOCTL_ISTIONAMESPACE=external-istiod
export ISTIOCTL_XDS_ADDRESS=external-f440ce970f6a672b5ae4eddf79834feb-0001.tor01.containers.appdomain.cloud:15012
```
After that, the following commands seemed to be mostly working properly:
```
istioctl analyze
istioctl validate
istioctl kube-inject
istioctl proxy-config
istioctl proxy-status
istioctl version
istioctl x describe
istioctl x internal-debug
```

**Comment by frankbu** [comment]
Fixed by #35746 #35675 and #35656
